specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/compute.json

```diff
@@ -7657,6 +7661,28 @@
     },
     "RestorePoint": {
       "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
```

Comment:
just curious, why this property need x-ms-client-flatten true? this extension will impact sdk and cli generation

Reply:
@yungezz this is the pattern followed in the compute.json file

x-ms-client-flatten

https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md
This extension allows to flatten deeply nested payloads into a more user friendly object.


Can you please elaborate, what do you mean by 'impact sdk and cli generation'?
Does using  "x-ms-client-flatten"=true break something?

Reply:
Customers are not using RestorePoint yet, so we dont have to worry about breaking changes for restore points

Reply:
what I mean is the extension will cause different generation logic in sdk/cli genertor, thus different interface.

Reply:
still not sure I understand, sorry.

Are you saying "x-ms-client-flatten"=true should never be used?
Am I using "x-ms-client-flatten"=true incorrectly?
Are there some cases, where it can be used, and other cases where it should not be used? What are those cases?